hawkmoth/parser.py

```diff
@@ -413,7 +421,23 @@ def parse(filename, domain='c', clang_args=None):
         result.add_child(ds)
 
     for cursor in tu.cursor.get_children():
-        if cursor.hash in comments:
+
+        # Identify `extern "C"` and `extern "C++"` blocks and recursively parse
+        # their contents. Only `extern "C"` is of any relevance in choosing a
+        # different domain.
+        # For some reason, the Python bindings don't return the cursor kind
+        # LINKAGE_SPEC as one would expect, so we need to do it the hard way.
+        if cursor.kind == CursorKind.UNEXPOSED_DECL:
+            tokens = cursor.get_tokens()
+            if next(tokens).spelling == 'extern':
```

Comment:
Apparently there are some `UNEXPOSED_DECL` cursors that have 0 tokens. The 1st case I found was `__BEGIN_DECLS` and I didn't check for others. These showed up in some standard headers and I'm guessing they're implementation specific stuff which are meaningless syntax wise.

Anyway, I did not account for such ghost cursors here and this naturally crashes when `next` fails to return a valid token.

Regardless of ghost cursors, assuming any cursor would not only have at least 1 token, but 2 was a mistake, so a similar fix is needed for the following `next` as well.

I actually have a fix already, but I'm avoiding force pushes for now.